### PR TITLE
UP-4854: Setup jacoco test coverage and coveralls reporting

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] tests are included

##### Description of change

https://issues.jasig.org/browse/UP-4854

:notebook: Jacoco can correctly handle Java 8 code, however it will skip subprojects that do not have any tests.

:warning: Coveralls Gradle logs the entire tests results to console. This can causes Travis CI to fail because the log size exceeds 4Mb. Log occurs at `INFO` level, this PR moves logging to quiet to prevent `INFO` and `WARN` level messages from appearing.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Sample report https://coveralls.io/builds/11851882

**Comment by bjagg** [comment]
Will test once conflicts are resolved..

**Comment by ChristianMurphy** [comment]
> Does this remove the need for Maven?

The re-organization of uPortal removed the need for Maven.
https://github.com/Jasig/uPortal/pull/916 removes all remaining references.

> Will test once conflicts are resolved..

Resolved.

File: build.gradle [hunk 1]
```diff
@@ -118,3 +124,28 @@ subprojects {
         }
     }
 }
+
+task jacocoAggregateReport(type: org.gradle.testing.jacoco.tasks.JacocoReport) {
```
Comment (ChristianMurphy) [inline build.gradle]: :notebook: I tried moving this to its own file. The use of the `subprojects` object prevents it from being used outside of the same file as where `subprojects` are declared.

File: .travis.yml [hunk 2]
```diff
@@ -2,17 +2,10 @@ language: java
...
-# ./gradlew assemble (default install command) won't cut it
-install:
-  - ./gradlew install
-  - mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
+install: ./gradlew install
 
-script:
-  - mvn clean test cobertura:cobertura coveralls:report -B -V
-  - ./gradlew -u -i -S --scan check
+script: ./gradlew -u -i -q -S --scan check jacocoAggregateReport coveralls
```
Comment (ChristianMurphy) [inline .travis.yml]: the quiet flag appears to be suppression the output from the scan flag. :disappointed:

**Comment by ChristianMurphy** [comment]
@apetro Would you be so kind as to re-enable to coveralls commit status?

**Pull request merged**